jwala-services/src/main/java/com/cerner/jwala/service/app/impl/ApplicationServiceImpl.java

```diff
@@ -132,18 +144,34 @@ public Application getApplication(final String name) {
     @Override
     public Application updateApplication(UpdateApplicationRequest updateApplicationRequest, User anUpdatingUser) {
         updateApplicationRequest.validate();
-
+        Application oldApplication = applicationPersistenceService.getApplication(updateApplicationRequest.getId());
         final Application application = applicationPersistenceService.updateApplication(updateApplicationRequest);
+
+        String appName = application.getName();
+        Identifier<Group> newGroupId = application.getGroup().getId();
+        Long id = newGroupId.getId();
+
+        List<Long> idList = new ArrayList<>();
+        idList.add(id);
+        List<JpaGroup> jpaGroups = groupPersistenceService.findGroups(idList);
+        if (jpaGroups.size() > 0) {
```

Comment:
Check specifically for one group and throw an exception otherwise. Our model only supports one group per application.